DrawerKit/DrawerKit/Internal API/PresentationController+Gestures.swift

```diff
@@ -15,7 +15,7 @@ extension PresentationController {
         guard tapY < currentDrawerY else { return }
         NotificationCenter.default.post(notification: DrawerNotification.drawerExteriorTapped)
         tapGesture.isEnabled = false
-        animateTransition(to: .collapsed)
+        self.presentedViewController.dismiss(animated: true)
```

Comment:
If I remember correctly, `animateTransition(to:)` animates the corner radius and also allow animations in the presented and presenting view controllers to occur concurrently with the drawer animation. Simply invoking `self.presentedViewController.dismiss(animated: true)` breaks all of that.

Reply:
I can take a look at that but I'd expect UIKit to call custom presentation machinery in this case instead of making us to perform animations manually.
Also from the point of this tap-to-dismiss functionality this is what it should do, dismiss, not animate to some state where it might be then dismiss or not due to collapsed state configuration.

Reply:
I think you're misunderstanding the situation. DK allows the client to tie animations in either or both the presenting and presented VCs with the animation of the drawer itself going up or down. UIKit knows nothing about those client animations so it's the responsibility of DK to make sure that they happen concurrently with the drawer animation. If you simply dismiss the drawer without catering to those animations, then the presented VC will be dismissed without its animations happening (no big deal, one might argue, since the VC is being dismissed anyway) and the presenting VC will be dismissed without its animations happening, which **is** a big deal because then the presenting VC will be left in an unexpected state from the point of the client.

Reply:
OK, I'll revert the change. Though IMO there shouldn't be parallel animations mechanism as UIKit already provides transition callbacks and ability to animate alongside animations but I'll not argue these implementations details now.

Reply:
> UIKit already provides transition callbacks and ability to animate alongside animations  

Yes, and DK uses them.